protocol/message_persistence.go

```diff
@@ -2420,15 +2419,16 @@ func getMessagesAndCursorsFromScanRows(db sqlitePersistence, rows *sql.Rows) ([]
 		if msg, ok := messageIdx[message.ID]; !ok {
 			messageIdx[message.ID] = &message
 			cursors = append(cursors, cursor)
+			messageIDs = append(messageIDs, message.ID)
 		} else if discordMessage := msg.GetDiscordMessage(); discordMessage != nil {
 			msg.Payload = getUpdatedChatMessagePayload(discordMessage, message.GetDiscordMessage())
 			messageIdx[message.ID] = msg
 		}
 	}
 
 	var messages common.Messages
-	for _, message := range messageIdx {
-		messages = append(messages, message)
+	for _, messageID := range messageIDs {
+		messages = append(messages, messageIdx[messageID])
```

Comment:
Can you do the same for `getPinnedMessagesAndCursorsFromScanRows` please?

Reply:
And also `getMessagesFromScanRows`

Reply:
Sounds good! This will make my PR colorful 😉

Reply:
also getMessagesAndCursorsFromScanRows? found by searching SortByClock